kinto/core/resource/__init__.py

```diff
@@ -717,7 +717,8 @@ def _raise_400_if_invalid_id(self, record_id):
 
         :raises: :class:`pyramid.httpexceptions.HTTPBadRequest`
         """
-        if not self.model.id_generator.match(six.text_type(record_id)):
+        is_string = isinstance(record_id, six.text_type)
```

Comment:
I think the right fix would be to convert the id as a string rather than raising an error.